tests/e2e/common.go

```diff
@@ -2,74 +2,121 @@ package e2e
-metadata:
-  name: example-velero
-spec:
-  olm_managed: false
-  default_velero_plugins:
-  - aws
-  - openshift
-  - csi
-  backup_storage_locations:
-  - name: default
-    provider: aws
-    object_storage:
-      bucket: myBucket
-      prefix: "velero"
-    config:
-      region: us-east-1
-      profile: "default"
-    credentials_secret_ref:
-      name: cloud-credentials
-      namespace: oadp-operator
-  volume_snapshot_locations:
-  - name: default
-    provider: aws
-    config:
-      region: us-west-2
-      profile: "default"
-  enable_restic: true
-  velero_feature_flags: EnableCSI`
-
-func decodeYaml() *unstructured.Unstructured {
+func getDefaultVeleroConfig(namespace string, s3Bucket string, credSecretRef string, instanceName string) *unstructured.Unstructured {
+	// Velero Instance creation spec with backupstorage location default to AWS. Would need to parameterize this later on to support multiple plugins.
+	var veleroSpec = unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "konveyor.openshift.io/v1alpha1",
+			"kind":       "Velero",
+			"metadata": map[string]interface{}{
+				"name":      instanceName,
+				"namespace": namespace,
+			},
+			"spec": map[string]interface{}{
+				"olm_managed": false,
+				"default_velero_plugins": []string{
+					"aws",
+					"csix",
+					"openshift",
+				},
+				"backup_storage_locations": [](map[string]interface{}){
+					map[string]interface{}{
+						"config": map[string]interface{}{
+							"profile": "default",
+							"region":  "us-east-1",
+						},
+						"credentials_secret_ref": map[string]interface{}{
+							"name":      credSecretRef,
+							"namespace": "oadp-operator",
+						},
+						"object_storage": map[string]interface{}{
+							"bucket": s3Bucket,
+							"prefix": "velero",
+						},
+						"name":     "default",
+						"provider": "aws",
+					},
+				},
+				"velero_feature_flags": "EnableCSI",
+				"enable_restic":        true,
+				"volume_snapshot_locations": [](map[string]interface{}){
+					map[string]interface{}{
+						"config": map[string]interface{}{
+							"profile": "default",
+							"region":  "us-west-2",
+						},
+						"name":     "default",
+						"provider": "aws",
+					},
+				},
+			},
+		},
+	}
+	return &veleroSpec
+}
+
+func decodeYaml(defaultVeleroConfigYAML string) *unstructured.Unstructured {
 	// set new unstructured type for Velero CR
 	unstructVelero := &unstructured.Unstructured{}
 
 	// decode yaml into unstructured type
 	dec := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
-	_, _, err := dec.Decode([]byte(DefaultVeleroConfigYAML), nil, unstructVelero)
+	_, _, err := dec.Decode([]byte(defaultVeleroConfigYAML), nil, unstructVelero)
 	if err != nil {
 		panic(err)
 	}
 	return unstructVelero
 }
 
-func createOADPTestNamespace() error {
+func getJsonData(path string) []byte {
+	// Return buffer data for json
+	jsonData, err := ioutil.ReadFile(path)
+	if err != nil {
+		panic(err)
```

Comment:
Rather than panicking, we should simply return an error and log it properly

Reply:
+1, `Expect().NotTo()` I think provides the trace log as well. So we might not need to use a logger also.

Reply:
Got it. Will return the error back and clean the code.

Reply:
Returning errors back for every utility function.